tools/nnyeah/nnyeah/MethodTransformations.cs

```diff
@@ -446,6 +447,20 @@ public class MethodTransformations {
 
 			allTransforms.Add (NewFuncXform ("System.TypeCode System.nfloat::GetTypeCode()", moduleToEdit, nfloatTypeRef, typeCodeRef, "GetTypeCode"));
 
+			var getConstantRef = GetGetConstantRef (modules.MicrosoftModule, moduleToEdit);
+
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_Video()", 0));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_Audio()", 1));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_Text()", 2));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_ClosedCaption()", 3));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_Subtitle()", 4));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_Timecode()", 5));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_TimedMetadata()", 6));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_Muxed()", 7));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_MetadataObject()", 8));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_Metadata()", 9));
+			allTransforms.Add (AVMediaTypeTransform (getConstantRef, "Foundation.NSString AVFoundation.AVMediaType::get_DepthData()", 10));
```

Comment:
Would it be possible to read these mappings from an xml file? So our users can provide their own

Reply:
Yes but it's a significant engineering effort. The problem is that, in this case the transform is very specific to the problem:
find a reference to a particular method and replace it with a specific set of IL instructions using a reference to a method looked up in the platform assembly.
The problem with this now is two-fold:
1. We would be writing an assembler using XML as a carrier syntax
2. We are providing the means to arbitrarily patch other people's compiled code, which is potentially a security problem. For example, let's say that there is a rule for "Find all calls to `bool SomeCommericalAssembly.CheckLicense ()` and replace them with `nop`" or some such thing. Another one would be "Find all calls to `void SomeAssembly.Authenticate (string userName, string password)` with:
```
stloc myPassword
stloc myUser
ldloc myUser
ldloc myPassword
call void SomeHorribleClass.RedirectUserAndPasswordToWarezSite(string, string)
ldloc myUser
ldloc myPassword
call bool void SomeAssembly.Authenticate (string userName, string password)
```
This ignores the time that's necessary to verify that every CIL instruction is valid.
It's a big sticky mess. If we restrict it to simple mapping then we couldn't handle the case I just implemented. If we make it general enough to handle that case, then we're writing a the patch-o-tron 3000® which is probably not we should be doing.